src/wp-includes/rest-api/endpoints/class-wp-rest-abilities-list-controller.php

```diff
@@ -0,0 +1,335 @@
+<?php
+/**
+ * REST API list controller for Abilities API.
+ *
+ * @package WordPress
+ * @subpackage Abilities_API
+ * @since 6.9.0
+ */
+
+declare( strict_types = 1 );
+
+/**
+ * Core controller used to access abilities via the REST API.
+ *
+ * @since 6.9.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Abilities_List_Controller extends WP_REST_Controller {
+
+	/**
+	 * Default number of items per page for pagination.
+	 *
+	 * @since 6.9.0
+	 * @var int
+	 */
+	public const DEFAULT_PER_PAGE = 50;
+
+	/**
+	 * REST API namespace.
+	 *
+	 * @since 6.9.0
+	 * @var string
+	 */
+	protected $namespace = 'wp/v2';
+
+	/**
+	 * REST API base route.
+	 *
+	 * @since 6.9.0
+	 * @var string
+	 */
+	protected $rest_base = 'abilities';
+
+	/**
+	 * Registers the routes for abilities.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>[a-zA-Z0-9\-\/]+)',
+			array(
+				'args'   => array(
+					'name' => array(
+						'description' => __( 'Unique identifier for the ability.' ),
+						'type'        => 'string',
+						'pattern'     => '^[a-zA-Z0-9\-\/]+$',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all abilities.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @param WP_REST_Request<array<string, mixed>> $request Full details about the request.
+	 * @return WP_REST_Response Response object on success.
+	 */
+	public function get_items( $request ) {
+		$abilities = array_filter(
+			wp_get_abilities(),
+			static function ( $ability ) {
+				return $ability->get_meta_item( 'show_in_rest' );
+			}
+		);
+
+		// Filter by ability category if specified.
+		$category = $request->get_param( 'category' );
```

Comment:
```suggestion
		$category = $request['category'];
```

Core controllers should just index in.